Report/sections/2_design/operational_semantics.tex

```diff
@@ -15,19 +27,30 @@ \subsubsection*{Abstract Syntax}
 	\begin{tabular}{l l}
 		$a \in \textbf{ArithmExp}$ & $\text{Arithmetic expressions}$ \\ 
 		$n \in \textbf{Num}$ & $\text{Numerals}$ \\
-		$x \in \textbf{Var}$ & $\text{Variables}$ \\
+		$k \in \textbf{Var}$ & $\text{Variables}$ \\
 		$A_v \in \textbf{AssignV}$ & $\text{Variable assignments}$ \\
-		$D_v \in \textbf{DeclV}$ & $\text{Variable declarations}$ \\
+		$DV \in \textbf{DeclV}$ & $\text{Variable declarations}$ \\
 		$C_m \in \textbf{ContDecl}$ & $\text{Content declarations}$ \\
-		$S \in \textbf{Stm}$ & $\text{Statements}$
+		$S \in \textbf{Stm}$ & $\text{Statements}$\\
+		$DR \in \textbf{DecR}$ & $\text{Record declarations}$\\
+		$r_n \in \textbf{Rnames}$ & $\text{Record names}$ \\
+		$K \in \textbf{Gvar}$ & $\text{Generalized variables}$\\
+		$P \in \textbf{Gpnames}$ & $\text{Generalized procedure names}$
 	\end{tabular}
 
 	\vspace{4mm}
 	\textit{Formation rules}
 	\begin{align*}
-		S&::=x=a\mid \texttt{skip}\mid S_1;S_2\mid \texttt{repeat }\{S\}; \\
+		K&::=k\mid r.K \\
+		P&::=p\mid r.P \\
+		S&::=K:=x=a\mid \texttt{skip}\mid S_1;S_2\mid \texttt{repeat }\{S\}; \mid L_\text{BRACK} \; D_V \; D_P \; D_R \; S \; R_\text{BRACK}\mid call \; P\\
 		a&::=n\mid x\mid a_1+a_2\mid a_1-a_2\mid a_1*a_2\mid a_1/a_2\mid (a_1) \\
-		VList&::=[a]\mid [a_1, \ldots, a_n]\mid [\epsilon]
+		%VList&::=[a]\mid [a_1, \ldots, a_n]\mid [\epsilon] \\
+		D_V&::= x:=a; \; D_V \mid \epsilon \\
+		D_P&::= p \; S; \; D_P \mid \epsilon \\
```

Comment:
```suggestion
		D_P&::= P \; S; \; D_P \mid \epsilon \\
```

Reply:
Or was this intentional? Then we need a definition for p in the syntactic categories